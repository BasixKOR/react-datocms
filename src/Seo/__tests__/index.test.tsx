import * as React from "react";
import { shallow } from "enzyme";
import { renderMetaTags, renderMetaTagsToString, ToMetaTagsType } from "../index";

const metaTags: ToMetaTagsType = [
  {
    attributes: null,
    content: "A new Media Area is online! - DatoCMS",
    tag: "title"
  },
  {
    attributes: {
      property: "og:title",
      content: "A new Media Area is online!"
    },
    content: null,
    tag: "meta"
  },
  {
    attributes: {
      name: "twitter:title",
      content: "A new Media Area is online!"
    },
    content: null,
    tag: "meta"
  }
];

describe("renderMetaTags", () => {
  it("generates an array of meta tags", () => {
    const wrapper = shallow(
      <head>
        {renderMetaTags(metaTags)}
      </head>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("renderMetaTagsToString", () => {
  it("generates an array of meta tags", () => {
    const result = renderMetaTagsToString(metaTags);
    expect(result).toEqual([
      '<title>A new Media Area is online! - DatoCMS</title>',
      '<meta property="og:title" content="A new Media Area is online!" />',
      '<meta name="twitter:title" content="A new Media Area is online!" />',
    ].join("\n"));
  });
});
