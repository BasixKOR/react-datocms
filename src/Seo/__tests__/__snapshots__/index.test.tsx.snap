// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderMetaTags generates an array of meta tags 1`] = `
<head>
  <title
    key="title"
  >
    A new Media Area is online! - DatoCMS
  </title>
  <meta
    content="A new Media Area is online!"
    key="meta-og:title"
    property="og:title"
  />
  <meta
    content="A new Media Area is online!"
    key="meta-twitter:title"
    name="twitter:title"
  />
</head>
`;
